src/inc/quic_platform_winuser.h

```diff
@@ -368,10 +375,7 @@ CxPlatPoolInitialize(
     _Inout_ CXPLAT_POOL* Pool
     )
 {
-#if DEBUG
-    CXPLAT_DBG_ASSERT(Size >= sizeof(CXPLAT_POOL_ENTRY));
-#endif
-    Pool->Size = Size;
+    Pool->Size = Size + sizeof(CXPLAT_POOL_HEADER); // Add space for the object header
```

Comment:
Do we need to check for overflows? Pool size is usually hardcoded so it is very unlikely, but this could be allocating less than requested and lead to write overflow (for a pool of 4GB buffers... ^^)

Reply:
Generally the sizes are small for pool allocations because you're expected to make a LOT of them. So, no, I don't think we need anything here.